src/components/pages/todo/TodoList.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { ChangeEvent, useState, useEffect } from 'react';
+import Checkbox from '@mui/material/Checkbox';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import Grid from '@mui/material/Grid';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import { DatePicker } from '@mui/x-date-pickers';
+import { format } from 'date-fns';
+import { Task } from '../../../interface/interface';
+import { styled } from '@mui/system';
+
+const defaultPlantList =
```

Comment:
i would separate constant into separate folder with constants, 
the functions that are not using state and just do some checks can be separated in utils folder, to keep files cleaner

Reply:
as I mentioned, I did not decompose this feature yet
decomposing all the stuff from the start would involve creating a ton of files
hope I'll come up with a better structure in future tasks